container/go/cmd/digester/digester.go

```diff
@@ -88,32 +93,92 @@ func main() {
 		}
 	}
 
-	img, err := utils.ReadImage(imgSrc, *format)
+	img, err := utils.ReadImage(imgSrc, *format, *tarball, layers)
 	if err != nil {
 		log.Fatalf("Error reading from %s: %v", imgSrc, err)
 	}
 
-	digest, err := img.Digest()
-	if err != nil {
-		log.Fatalf("Error getting image digest: %v", err)
+	configPath := ""
+	if *format == "legacy"{
+		configPath = *src
 	}
-
-	err = writeDigest(digest, *dst)
+	err = writeDigest(img, *dst, configPath)
 	if err != nil {
 		log.Fatalf("Error outputting digest file to %s: %v", *dst, err)
 	}
 
 	log.Printf("Successfully generated image digest file at %s", *dst)
 }
 
-// writeDigest outputs digest to a "digest file" at dst. Digest is the image manifest's sha256 hash.
-func writeDigest(digest v1.Hash, dst string) error {
+// writeDigest outputs image's digest to a "digest file" at dst. Digest is the image manifest's sha256 hash.
+func writeDigest(image v1.Image, dst, configPath string) error {
+	m, err := image.Manifest()
+	if err != nil {
+		return errors.Wrap(err, "error getting image manifest")
+	}
+
+	if configPath != "" {
+		rawConfig, err := ioutil.ReadFile(configPath)
+		if err != nil {
+			return errors.Wrap(err, "Error getting image raw config")
+		}
+		
+		cfgHash, cfgSize, err := v1.SHA256(bytes.NewReader(rawConfig))
+		if err != nil {
+			return errors.Wrap(err, "unable to hash image config file")
+		}
+
+		m.Config = v1.Descriptor{
+			MediaType: types.DockerConfigJSON,
+			Size:      cfgSize,
+			Digest:    cfgHash,
+		}
+	}
+
+	out, err := formatStruct(m)
+	if err != nil {
+		return err
+	}
+
+	digest, _, err := v1.SHA256(bytes.NewReader((*out).Bytes()))
+	if err != nil {
+		return errors.Wrap(err, "unable to compute sha256 sum for digest")
+	}
+
 	rawDigest := []byte(digest.Algorithm + ":" + digest.Hex)
 
-	err := ioutil.WriteFile(dst, rawDigest, os.ModePerm)
+	err = ioutil.WriteFile(dst, rawDigest, os.ModePerm)
 	if err != nil {
 		return errors.Wrapf(err, "unable to write digest file to %s", dst)
 	}
 
 	return nil
 }
+
+func formatStruct(s interface{}) (*bytes.Buffer, error) {
```

Comment:
Add function comment. Also, this function doesn't need to be this generic as it's only being used to format the manifest. I would suggest naming it `manifestToJSON` and accepting a `v1.Manifest` as the input parameter.